python_modules/dagster/dagster/core/storage/event_log/base.py

```diff
@@ -166,21 +172,41 @@ class EventLogStorage(ABC, MayHaveInstanceWeakref):
     should be done by setting values in that file.
     """
 
-    @abstractmethod
     def get_logs_for_run(
         self,
         run_id: str,
-        cursor: Optional[int] = -1,
+        cursor: Optional[Union[str, int]] = None,
         of_type: Optional[Union[DagsterEventType, Set[DagsterEventType]]] = None,
         limit: Optional[int] = None,
     ) -> Iterable[EventLogEntry]:
         """Get all of the logs corresponding to a run.
 
         Args:
             run_id (str): The id of the run for which to fetch logs.
-            cursor (Optional[int]): Zero-indexed logs will be returned starting from cursor + 1,
-                i.e., if cursor is -1, all logs will be returned. (default: -1)
+            cursor (Optional[Union[str, int]]): Cursor value to track paginated queries.  Legacy
+                support for integer offset cursors.
+            of_type (Optional[DagsterEventType]): the dagster event type to filter the logs.
+            limit (Optional[int]): Max number of records to return.
+        """
+        records = self.get_records_for_run(run_id, cursor, of_type, limit).records
+        return [record.event_log_entry for record in records]
```

Comment:
<!-- GRAPHITE_METADATA {"trueLocation":{"startLine":175,"endLine":192,"prLink":"https://app.graphite.dev/github/pr/dagster-io/dagster/7973"}} -->
<i class='graphite__hidden'>[Re: lines 175 to 192]<br/></i>

should we support new capability in old API or just migrate to new one? <span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/7973?utm_source=unchanged-line-comment">Graphite</a>.</span>